File: napa/TripViewController.m
//
//  TripViewController.m
//  napa
//
//  Created by Remy JOURDE on 11/08/2014.
//  Copyright (c) 2014 Remy Jourde. All rights reserved.
//

#import "TripViewController.h"

@interface TripViewController ()

@end

@implementation TripViewController

#pragma mark - Managing the trip

- (void)setTrip:(Trip*)newTrip
{
    if (_trip != newTrip) {
        _trip = newTrip;
        
        // Update the view.
        [self configureView];
    }
}

- (void)configureView
{
    // Update the user interface for the detail item.
    if (self.trip) {
        self.navigationItem.title = self.trip.name;
    }
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: napa/TripViewController.swift
//
//  TripViewController.swift
//  napa
//
//  Created by Remy JOURDE on 11/10/2014.
//  Copyright (c) 2014 Remy Jourde. All rights reserved.
//

import UIKit

class TripViewController: UITabBarController {

    var trip: Trip?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - Managing the trip
    
    func setTrip(newTrip: Trip) {
        if self.trip != newTrip {
            self.trip = newTrip
    
            // Update the navigation title
            self.navigationItem.title = self.trip?.name
        }
    }

}
